.github/workflows/tests.yml

```diff
@@ -26,25 +26,45 @@ jobs:
           - { name: "3.9", python: "3.9", os: ubuntu-latest, tox: py39 }
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
       - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python }}
-      - name: Update pip
+      - name: Install poetry
+        run: |
+          python -m pip install poetry
+      - name: Configure poetry
+        run: |
+          python -m poetry config virtualenvs.in-project true
+      - name: Cache the virtualenv
+        uses: actions/cache@v3
+        with:
+          path: ./.venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('pyproject.toml') }}
+      - name: Install poetry plugin
+        run: python -m poetry self add "poetry-dynamic-versioning[plugin]"
+      # Install test dependencies (tox) to the root (non-package install)
+      - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-      - name: Get pip cache dir
-        id: pip-cache
-        run: echo "dir=$(pip cache dir)" >> $GITHUB_OUTPUT
-      - uses: actions/cache@v3
-        with:
-          path: ${{ steps.pip-cache.outputs.dir }}
-          key: pip|${{ runner.os }}|${{ matrix.python }}|${{ hashFiles('pyproject.toml') }}
+          python -m poetry install --no-root --only test
       - name: set full Python version in PY env var
         # See https://pre-commit.com/#github-actions-example
         run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
-      - run: python -m pip install tox
-      - run: python -m tox -e ${{ matrix.tox }}
+      # Install node and yarn in order to build the front end during packaging  
+      - name: Set Node.js 18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+      - name: Install Yarn
+        run: npm install -g yarn
```

Comment:
Think this should be fine but there's also https://github.com/marketplace/actions/github-action-for-yarn, which might be faster / more reliable

Reply:
I think that group of Actions is for performing `yarn <thing>` commands mostly - I didn't see anything in there or the node installation Action about installing `yarn` itself 🤔 

This is because you need it available on the Runner's path, for when `yarn` is called in the shell (when running a build, while in `tox` or similar)

I did see [some discussion](https://github.com/actions/setup-node/issues/182) on this on the node Action but it's the usual descent into madness

Reply:
Yeah than let's leave it as is, as I said I think this should be fine :)